### ISSUE TITLE
clickhouse version 21.8 data cannot move Part

### ISSUE THREAD

**Issue by sunny19930321** [ClickHouse/ClickHouse#33423]
> Make sure to check documentation https://clickhouse.yandex/docs/en/ first. If the question is concise and probably has a short answer, asking it in Telegram chat https://telegram.me/clickhouse_en is probably the fastest way to find the answer. For more complicated questions, consider asking them on StackOverflow with "clickhouse" tag https://stackoverflow.com/questions/tagged/clickhouse 

> If you still prefer GitHub issues, remove all this text and ask your question here.

**Comment by alexey-milovidov** [comment]
Upgrade to version 21.12.

**Comment by sunny19930321** [comment]
> Upgrade to version 21.12.

yes, but old data cannot be move part  because the part does not have a uuid

**Comment by nvartolomei** [comment]
It looks like you are using a tool to automatically move parts? Curious what it is? Is it on GitHub? Wondering if it is missing some documentation.

Parts don’t have uuids by default yet. I haven’t figured out how to enable this by default without a breaking change yet.

First you need to enable the assign part uuids setting in clickhouse config and then new parts will have uuids and will become movable. To force assignment of uuids to old parts you need to trigger a mutation on the table and wait for it to recreate the parts. The mutation could be something really simple like ‘update set col = col where 1 = 1’. I think a simple ‘alter add column’ could work as well, don’t remember where this recreates parts today.

**Comment by nvartolomei** [comment]
Duplicate #33428

**Comment by sunny19930321** [comment]
> It looks like you are using a tool to automatically move parts? Curious what it is? Is it on GitHub? Wondering if it is missing some documentation.
> 
> Parts don’t have uuids by default yet. I haven’t figured out how to enable this by default without a breaking change yet.
> 
> First you need to enable the assign part uuids setting in clickhouse config and then new parts will have uuids and will become movable. To force assignment of uuids to old parts you need to trigger a mutation on the table and wait for it to recreate the parts. The mutation could be something really simple like ‘update set col = col where 1 = 1’. I think a simple ‘alter add column’ could work as well, don’t remember where this recreates parts today.

yes, our internal elastic scaling tool based on Move Part has not been open source on Github yet, and the above problem has been generated by the new uuid through optimize

**Comment by nvartolomei** [comment]
Are you able to reproduce the issue? Can you provide an example?

**Comment by sunny19930321** [comment]
(1) 21.8 verstion part infomation
```
SELECT
    partition,
    name,
    uuid
FROM system.parts
WHERE table = 'behavior2_view_local'
LIMIT 1

Query id: ece0c5f5-f2bf-40c1-b1e7-9654f34bb434

┌─partition──┬─name────────────────────────────────────┬─uuid─────────────────────────────────┐
│ 2021-11-21 │ 02d5c9d43e33d0ac52b867c40434a8de_0_19_2 │ 00000000-0000-0000-0000-000000000000 │
└────────────┴─────────────────────────────────────────┴──────────────────────────────────────┘

1 rows in set. Elapsed: 0.007 sec.
```

(2) upgrade version to 21.9, use move part function
```
ALTER TABLE label.behavior2_view_local
    MOVE PART '02d5c9d43e33d0ac52b867c40434a8de_0_19_2' TO '/clickhouse/tables/label/behavior2_view_local/013'

Query id: d887a913-330b-40bc-8d22-13ce2184c0d1


0 rows in set. Elapsed: 0.006 sec.

Received exception from server (version 21.9.1):
Code: 48. DB::Exception: Received from 172.21.68.72:9600. DB::Exception: Part 02d5c9d43e33d0ac52b867c40434a8de_0_19_2 does not have an uuid assigned and it can't be moved between shards.
```

**Comment by sunny19930321** [comment]
When moving a part that does not have a UUID, reassign a UUID，because Optimze or mutation, performance is very slow
